assets/javascripts/discourse/models/filter-rule.js.es6

```diff
@@ -1,17 +1,22 @@
 import RestModel from 'discourse/models/rest';
 
 export default RestModel.extend({
-  category_id: -1,
+  category_id: '',
   channel: '',
   filter: null,
 
   category: function() {
     var id = this.get('category_id');
 
-    if (id === "0")
-      return Discourse.Category.create({ name: 'All Categories', id: 0 });
-    else {
-      return Discourse.Category.findById(id) || { id: id, name: 'Deleted Category' };
+    switch (id) {
+      case '*':
+        return Discourse.Category.create({ name: I18n.t('slack.choose.all_categories'), id: '*' });
+        break;
+      case '':
```

Comment:
@tgxworld since my this PR gone too much in wrong direction I recreated a new [PR](https://github.com/discourse/discourse-slack-official/pull/22) with `AND` functionality.

Now I am using same old way to maintain the data. Just added the tags support additionally :+1: